File: frontend/lib/domain/entities/album.dart
import 'package:equatable/equatable.dart';

class Album extends Equatable {
  final String id;
  final String title;
  final String artistId;
  final String artistName;
  final String? coverUrl;
  final int totalTracks;
  final DateTime releasedAt;

  const Album({
    required this.id,
    required this.title,
    required this.artistId,
    required this.artistName,
    this.coverUrl,
    this.totalTracks = 0,
    required this.releasedAt,
  });

  @override
  List<Object?> get props => [id, title, artistId];
}

File: backend/src/main/java/com/kaemusic/backend/data/entities/Album.java
package com.kaemusic.backend.data.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "albums")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Album {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "artist_id", nullable = false)
    private Artist artist;

    private String coverUrl;
    private int totalTracks;
    private LocalDateTime releasedAt;
}
